src/plugins/graph/models/data-configuration-model.ts

```diff
@@ -588,18 +610,17 @@ export const DataConfigurationModel = types
       self.setPointsNeedUpdating(true);
     },
     setDataset(dataset: IDataSet | undefined, metadata: ISharedCaseMetadata | undefined) {
-      self.actionHandlerDisposer?.();
-      self.actionHandlerDisposer = undefined;
       self.dataset = dataset;
       self.metadata = metadata;
-      self.filteredCases = [];
-      if (dataset) {
+      this.handleDataSetChange();
+    },
+    handleDataSetChange() {
+      self.actionHandlerDisposer?.();
+      self.actionHandlerDisposer = undefined;
+      self.filteredCases.clear();
+      if (self.dataset) {
```

Comment:
Note how this was moved into a MobX reaction in https://github.com/concord-consortium/codap/pull/971.